executor/logging.go

```diff
@@ -2,23 +2,129 @@ package executor
 
 import (
 	"bufio"
+	"fmt"
 	"io"
 	"log"
+	"time"
 )
 
+var stdLogBufferSize = 64 * 1024
+
 // bindLoggingPipe spawns a goroutine for passing through logging of the given output pipe.
+//
+// This implementation is adapted from https://github.com/hashicorp/go-plugin/pull/98
 func bindLoggingPipe(name string, pipe io.Reader, output io.Writer) {
 	log.Printf("Started logging %s from function.", name)
 
-	scanner := bufio.NewScanner(pipe)
-	logger := log.New(output, log.Prefix(), log.Flags())
+	scanner := bufio.NewReaderSize(pipe, stdLogBufferSize)
 
 	go func() {
-		for scanner.Scan() {
-			logger.Printf("%s: %s", name, scanner.Text())
-		}
-		if err := scanner.Err(); err != nil {
-			log.Printf("Error scanning %s: %s", name, err.Error())
+		logFlags := log.Flags()
+		continuation := false
+
+		for {
+			line, isPrefix, err := scanner.ReadLine()
+			switch {
+			case err == io.EOF:
+				return
+			case err != nil:
+				log.Printf("Error scanning %s: %s", name, err.Error())
+				return
+			}
+
+			fmt.Println(isPrefix, string(line))
```

Comment:
Is this the equivalent of:

```
			logger.Printf("%s: %s", name, scanner.Text())		
```

Or a debug line left over?

Reply:
Yes, i will remove it